src/index.ts

```diff
@@ -113,4 +114,29 @@ export class AtlasIndex extends BaseAtlasClass {
     )) as Table;
     return tb;
   }
+
+  /**
+   *
+   * @param param0 A keyed dictionary including `k` (the number of neighbors to return)
+   * and `queries` (a list of vectors to search for).
+   * @returns
+   */
+  async nearest_neighbors_by_vector({
+    k = 10,
+    queries,
+  }: Omit<
+    components['schemas']['EmbeddingNeighborRequest'],
+    'atlas_index_id'
+  >): Promise<components['schemas']['EmbeddingNeighborResponse']> {
+    const { neighbors, distances } = (await this.apiCall(
+      `/v1/project/data/get/nearest_neighbors/by_embedding`,
+      'POST',
+      {
+        atlas_index_id: this.id,
+        k,
+        queries,
+      }
+    )) as components['schemas']['EmbeddingNeighborResponse'];
```

Comment:
Instead of doing an 'as' and force type checking. Let's use 'satisfies'

Reply:
If you can show a way that this works please do, I don't know what you mean here.

Reply:
The issue is that API call can return many different forms, so satisfies fails type checking.